src/util/alloc/bumpallocator.rs

```diff
@@ -18,9 +18,9 @@ pub struct BumpAllocator<VM: VMBinding> {
     /// [`VMThread`] associated with this allocator instance
     pub tls: VMThread,
     /// Current cursor for bump pointer
-    cursor: Address,
+    pub(crate) cursor: Address,
     /// Limit for bump pointer
-    limit: Address,
+    pub(crate) limit: Address,
```

Comment:
We can make a `#[repr(C)] struct` for these two fields since it appears so often.  Then we only need to return one offset for those two fields.

Reply:
Should this be done in this PR as well?

Reply:
Yes, please.  This is the best chance to do it.

Reply:
How this structure should be named?

Reply:
Good question.  I think we should discuss on Zulip and let everyone vote.

It should be defined in `allocator.rs` because it is common to many concrete allocators.

Reply:
1. This PR seems a step in the right direction.  Thank you!
2. Placing cursor and limit in a struct seems sensible.
3. The cursor and limit should be called a bump pointer, because that is literally what it is.  I teach people how GCs are built and this is literally how I describe a bump pointer: a cursor and a limit.  Inventing a new term to describe this most basic of data structures is not a good idea.

Meta point: I don't think "voting" on conceptual points like this on Zulip is a great idea in general.